ribasim_qgis/core/topology.py

```diff
@@ -29,7 +37,9 @@ def explode_lines(edge: QgsVectorLayer) -> None:
     return
 
 
-def derive_connectivity(node_index, node_xy, edge_xy) -> tuple[np.ndarray, np.ndarray]:
+def derive_connectivity(
+    node_index, node_xy, edge_xy
+) -> tuple[np.ndarray[Any, Any], np.ndarray[Any, Any]]:
```

Comment:
I would like to get rid of these `Any`s. Do we know the types of these objects?

Reply:
These are both 1D integer arrays.

Reply:
Also `numpy.typing.NDArray` is more useful here, it's seldom useful to specify the shape.
Here that would be `NDArray[np.int_]`